lib/stat_tracker.rb

```diff
@@ -598,5 +607,73 @@ def fewest_tackles(season_id)
   end
 
   ######################END of SEASON STATS####################
-  
+  # start of sienna's league stats
+  def home_game_teams
+    game_team_collection.all.find_all do |game_team|
+      game_team.hoa == "home"
+    end
+  end
+
+  def home_game_teams_by_team
+    home_game_teams.group_by do |game_team|
+      game_team.team_id
+    end
+  end
+
+  def total_home_goals_grouped_by_team
+    goals_by_team_id = Hash.new(0)
+    home_game_teams_by_team.each do |team_id, game_team_coll|
+      game_team_coll.each do |game_team|
+        goals_by_team_id[team_id] += game_team.goals.to_i
+      end
+    end
+    goals_by_team_id
+  end
+
+  def total_home_games_grouped_by_team
+    games_by_team_id = Hash.new(0)
+    home_game_teams_by_team.each do |team_id, game_team_coll|
+      game_team_coll.each do |game_team|
+        games_by_team_id[team_id] += 1
+      end
+    end
+    games_by_team_id
+  end
+
+  def ratio_home_goals_to_games_grouped_by_team
+    ratio_goals_to_games_by_team = Hash.new(0)
+    total_home_games_grouped_by_team.each do |games_team_id, total_games|
+      total_home_goals_grouped_by_team.each do |goals_team_id, total_goals|
+        if games_team_id == goals_team_id
+          ratio_goals_to_games_by_team[games_team_id] = (total_goals.to_f / total_games.to_f)
+        end
+      end
+    end
+    ratio_goals_to_games_by_team
```

Comment:
Good way to combine both home and away teams to get the final team with their ratio.